lightning/src/ln/channelmanager.rs

```diff
@@ -474,7 +493,7 @@ pub struct ChannelManager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref,
 	/// after reloading from disk while replaying blocks against ChannelMonitors.
 	///
 	/// Locked *after* channel_state.
-	pending_outbound_payments: Mutex<HashSet<[u8; 32]>>,
+	pending_outbound_payments: Mutex<HashSet<([u8; 32], Option<MppId>)>>,
```

Comment:
I have to admit this datastructure "feels" wrong - you have to walk the full thing to remove all payments with a given mpp id, which is maybe ok, but if we also want to change the event types based on "are there pending htlcs with the same mpp id" we probably need it to be a map from MppId to the per-payment info. Of course that also sucks cause then we either have a multi-layer `HashMap` or a ton of entries for the no-MppId key. Another option would be a something b-tree based, which would work here, but Rust's BTreeSet doesn't support `drain(range)`, so doesn't quite have what we need.

Maybe we should do a `HashMap<MppId, Option<HashSet<[u8; 32]>>>` - always set an MppId even for non-MPP payments, and only fill in the second set if its MPP. I hate multi-level maps like that, but I'm not sure any alternative is better, and walking the map sucks too.